packages/front-end/components/Settings/CreditCardModal.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+import {
+  PaymentElement,
+  useElements,
+  useStripe,
+} from "@stripe/react-stripe-js";
+import { Flex, Text } from "@radix-ui/themes";
+import { useAuth } from "@/services/auth";
+import Modal from "../Modal";
+import { useStripeContext } from "../Billing/StripeProviderWrapper";
+import Toggle from "../Forms/Toggle";
+
+interface Props {
+  onClose: () => void;
+  refetch: () => void;
+}
+
+export default function CreditCardModal({ onClose, refetch }: Props) {
+  const [defaultCard, setDefaultCard] = useState(true);
+  const { clientSecret } = useStripeContext();
+  const { apiCall } = useAuth();
+  const elements = useElements();
+  const stripe = useStripe();
+
+  const handleSubmit = async () => {
+    if (!stripe || !elements || !clientSecret) return;
+    try {
+      // Trigger form validation and wallet collection
+      const { error: submitError } = await elements.submit();
+      if (submitError) {
+        throw new Error(
+          submitError.message || "Unable to validate card inputs"
+        );
+      }
+
+      const { setupIntent } = await stripe.confirmSetup({
+        elements,
+        clientSecret,
+        confirmParams: {
+          return_url: `${process.env.NEXT_PUBLIC_API_HOST}/settings/billing`,
```

Comment:
Were you able to test this?  When redirecting back to the billing page, I think the modal would not be open anymore.  I guess if it added the card to the customer correctly, it would start showing up in the list of payment methods and they could mark it as default there instead, so not too bad.